c7n/resources/s3.py

```diff
@@ -980,18 +981,22 @@ class SetPolicyStatement(BucketActionBase):
                     },
                     'required': ['Sid', 'Effect'],
                     'oneOf': [
-                        {'required': ['Action', 'Resource']},
-                        {'required': ['NotAction', 'Resource']},
-                        {'required': ['Action', 'NotResource']},
-                        {'required': ['NotAction', 'NotResource']}
+                        {'required': ['Principal', 'Action', 'Resource']},
+                        {'required': ['NotPrincipal', 'Action', 'Resource']},
+                        {'required': ['Principal', 'NotAction', 'Resource']},
+                        {'required': ['NotPrincipal', 'NotAction', 'Resource']},
+                        {'required': ['Principal', 'Action', 'NotResource']},
+                        {'required': ['NotPrincipal', 'Action', 'NotResource']},
+                        {'required': ['Principal', 'NotAction', 'NotResource']},
+                        {'required': ['NotPrincipal', 'NotAction', 'NotResource']}
                     ]
                 }
             }
         }
     )
 
     def process_bucket(self, bucket):
-        policy = bucket.get('Policy', '{}')
+        policy = bucket['Policy'] or '{}'
```

Comment:
per previous comment in the pr, if custodian execution role does not have access to the bucket policy, this key won't be extant, denied augment methods skip values and populate the deniedmethods annotation, defaults are currently used for missing values.